### ISSUE TITLE
Ignore directives for imports-first

### ISSUE THREAD

**Issue by lemonmade** [benmosher/eslint-plugin-import#255]
We currently use a custom directive at the top of most JavaScript files. `imports-first` requires that any imports come before this line, but doing so means that it is no longer treated as a directive. I'm wondering if it would be possible to add an option to `imports-first` to ignore directives (which should be pretty easy to detect, even though I don't think ESLint exposes them as such). Basically, I would love something that lets the following pass:

```js
'expose Foo'

import MyThing from './somewhere'
// ...
```

but continue to fail anything that is not a directive appearing first:

```js
doSomething();

import MyThing from './somewhere'
// ...
```

I'm happy to take a stab at this if it's something you are interested in. Thanks!

**Comment by benmosher** [comment]
Ah, makes sense. Hadn't thought about that before considering modules imply strict mode. 

A PR allowing directives (which I guess is defined as an ExpressionStatement containing a string Literal) would be welcome. 😄

**Comment by lemonmade** [comment]
Cool, I can definitely do that. Do you want this as an option or as the default behaviour? Any preference for an option name if you want it as an option?

**Comment by benmosher** [comment]
I think it makes sense to always allow directives, until someone says otherwise.

I'm struggling to imagine why you'd want to report pragmas/directives, and if that day comes it's easy to add an option.
